src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -904,6 +904,7 @@ private string GetDebuggerDisplay()
                 // Use the binder that owns the scope for the local because this (the current) binder
                 // might own nested scope.
                 var hasErrors = localSymbol.ScopeBinder.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
+                ReportFieldOrValueContextualKeywordConflictIfAny(designation, designation.Identifier, diagnostics);
```

Comment:
>ReportFieldOrValueContextualKeywordConflictIfAny [](http://example.com/codeflow?start=16&length=48)

Consider placing this check at the beginning of the function. in other words close to the ```LookupLocal``` call.